packages/shared/src/components/streak/popup/ReadingStreakPopup.tsx

```diff
@@ -19,52 +23,74 @@ const streakDays = [
   addDays(today, 4),
 ]; // these dates will then be compared to the user's post views
 
-export function ReadingStreakPopup(): ReactElement {
-  return (
-    <div className="flex flex-col">
-      <div className="flex flex-row">
-        <StreakSection streak={44} label="Current streak" />
-        <StreakSection streak={100} label="Longest streak 🏆" />
-      </div>
-      <div className="mt-6 flex flex-row gap-2">
-        {streakDays.map((value) => {
-          const day = value.getDay();
-          const date = value.getDate();
+interface ReadingStreakPopupProps {
+  streak: UserStreak;
+}
+
+export function ReadingStreakPopup({
+  streak,
+}: ReadingStreakPopupProps): ReactElement {
+  const { user } = useAuthContext();
+  const { data: history } = useQuery(
+    generateQueryKey(RequestKey.ReadingStreak30Days, user),
+    () => getReadingStreak30Days(user.id),
+  );
 
-          // this is just a dummy logic to simulate the popup with different values
-          const getStreak = () => {
-            if (date === dateToday) {
-              return Streak.Pending;
-            }
+  const streaks = useMemo(() => {
+    return streakDays.map((value) => {
+      const day = value.getDay();
+      const date = value.getDate();
```

Comment:
Fun with timezones never ends. By default, the browser uses the timezone setting from the OS. This can potentially be different from the timezone the user has configured on their user profile.

I think we should be consistent with our calculation and need to convert all of these dates to the timezone the user specified on their user profile. [Since the calculation on the graphql endpoint is already doing that as well (see the `timestampAtTimezone`)](https://github.com/dailydotdev/daily-api/blob/mi-71-user-streak-counter/src/schema/users.ts#L843)

Reply:
The values being processed here are all the result of the BE's response. FE-wise, the checks here should be good to go.

Reply:
Let's try it out with different timezones in the future and see what happens 👍